canonical-web-and-design/maas-ui
# feat: remove /r path from maas-ui

## Done

- remove `/r` path from maas-ui
- add redirect from `/r` to `/MAAS`
- update controller details links to react
- remove `LegacyLink` component
- add `isControllerDetails` check to `ControllerVLANs` hook

Note: Includes changes from https://github.com/canonical-web-and-design/maas-ui/pull/4039 - it can be reviewed and merged separately first if you prefer

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- run `start` and `ui` scripts and verify all routes work correctly, in particular:
- Go to controller listing page and navigate to controller details page
- Verify the react component is displayed

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/959

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.
